layersvt/screenshot.cpp

```diff
@@ -364,18 +366,126 @@ static void writePPM(const char *filename, VkImage image1) {
     // Gather incoming image info and check image format for compatibility with
     // the target format.
     // This function supports both 24-bit and 32-bit swapchain images.
-    VkFormat const target32bitFormat = VK_FORMAT_R8G8B8A8_UNORM;
-    VkFormat const target24bitFormat = VK_FORMAT_R8G8B8_UNORM;
     uint32_t const width = imageMap[image1]->imageExtent.width;
     uint32_t const height = imageMap[image1]->imageExtent.height;
     VkFormat const format = imageMap[image1]->format;
     uint32_t const numChannels = vk_format_get_channel_count(format);
-    if ((vk_format_get_compatibility_class(target24bitFormat) != vk_format_get_compatibility_class(format)) &&
-        (vk_format_get_compatibility_class(target32bitFormat) != vk_format_get_compatibility_class(format))) {
+
+    if ((3 != numChannels) && (4 != numChannels)) {
         assert(0);
         return;
     }
-    if ((3 != numChannels) && (4 != numChannels)) {
+
+    // Initial dest format is undefined as we will look for one
+    VkFormat destformat = VK_FORMAT_UNDEFINED;
+
+    //Get users request is specific color space format required
+    const char *vk_screenshot_format = local_getenv(env_var_format);
+    if (vk_screenshot_format && *vk_screenshot_format) {
+        if (!strcmp(vk_screenshot_format, "UNORM")) {
+            if (numChannels == 4)
+                destformat = VK_FORMAT_R8G8B8A8_UNORM;
+           else
+                destformat = VK_FORMAT_R8G8B8_UNORM;
+        } else if (!strcmp(vk_screenshot_format, "SRGB")) {
+            if (numChannels == 4)
+                destformat = VK_FORMAT_R8G8B8A8_SRGB;
+            else
+                destformat = VK_FORMAT_R8G8B8_SRGB;
+        } else if (!strcmp(vk_screenshot_format, "SNORM")) {
+            if (numChannels == 4)
+                destformat = VK_FORMAT_R8G8B8A8_SNORM;
+            else
+                destformat = VK_FORMAT_R8G8B8_SNORM;
+        } else if (!strcmp(vk_screenshot_format, "USCALED")) {
+            if (numChannels == 4)
+                destformat = VK_FORMAT_R8G8B8A8_USCALED;
+            else
+                destformat = VK_FORMAT_R8G8B8_USCALED;
+        } else if (!strcmp(vk_screenshot_format, "SSCALED")) {
+            if (numChannels == 4)
+                destformat = VK_FORMAT_R8G8B8A8_SSCALED;
+            else
+                destformat = VK_FORMAT_R8G8B8_SSCALED;
+        } else if (!strcmp(vk_screenshot_format, "UINT")) {
+            if (numChannels == 4)
+                destformat = VK_FORMAT_R8G8B8A8_UINT;
+            else
+                destformat = VK_FORMAT_R8G8B8_UINT;
+        } else if (!strcmp(vk_screenshot_format, "SINT")) {
+            if (numChannels == 4)
+                destformat = VK_FORMAT_R8G8B8A8_SINT;
+            else
+                destformat = VK_FORMAT_R8G8B8_SINT;
+        } else {
+#ifdef ANDROID
+#else
+            fprintf(stderr, "Selected format:%s\nIs NOT in the list:\nUNORM, SNORM, USCALED, SSCALED, UINT, SINT, SRGB\n"
```

Comment:
This message should be printed only once. If multiple screenshots are done, it looks like it will be printed for each screenshot.